Apps/frontend/src/components/Waves.svelte

```diff
@@ -103,9 +122,9 @@
   };
 
   const update = () => {
-    for (let i = 0; i < channel_samples.length; i++) {
+    for (let i = 0; i < channelSamplesInjectCursor.length; i++) {
```

Comment:
I'm not happy with an extra for loop only to "write" the cursor, currently looking for a way to fix this, will let you know if I find a better solution.

Reply:
I replaced the old loop, because the update function doesn't know where we currently are, it's just plots all the data.
So I first inject the cursor when data arrives.